### PR TITLE
Additions to Linux diagnostic manifest for Azure Linux Escalation Team

### PR DESCRIPTION
**Pull request by peden-msft**

### CODE REVIEW

**Review by jasonzio — CHANGES_REQUESTED** [review]
This is a very large expansion of the total data collected by the "diagnostic" mode. I am very concerned that adding all this will result in more timeouts or "too large" zip bundles. I'd prefer to see most of the net-new file collections happen in new modes. A quick look shows there are at least two obvious segmentations:
- bootconfig
- pkgmgmt

I realize this means the on-call might need to run IID multiple times to get all the necessary data, but that's preferred over running it multiple times because of timeouts due to the size of data being collected.

File: pyServer/manifests/linux/diagnostic [hunk 1]
```diff
@@ -104,6 +110,7 @@ copy,/var/log/dmesg*
 copy,/var/log/boot*
 copy,/var/log/auth*
 copy,/var/log/secure*
+copy,/var/log/cloudregister
```
Comment (chintanrp) [inline pyServer/manifests/linux/diagnostic:113]: How big this file can be?
As well /var/log/zypp/history and /boot ?
Just need to make sure that we are not adding a big chunk as part of diagnostic manifest.
Also for all of these new addition, if you are sure for any of that that it does not contain any secret then we can add "noscan" to skip credscan and improve the performance.

**Comment by jasonzio** [comment]
You should monitor the impact of the expansion of the "diagnostic" mode to see if there's an increase in timeouts. (I'm not sure where the telemetry feed for IID lives, but the Geneva Actions team should be able to point you in the right direction.)

**Review by jasonzio — DISMISSED** [review]

**Comment by chintanrp** [comment]
> You should monitor the impact of the expansion of the "diagnostic" mode to see if there's an increase in timeouts. (I'm not sure where the telemetry feed for IID lives, but the Geneva Actions team should be able to point you in the right direction.)

@jasonzio  - Yes we can monitor that. But before that we just want to make sure about we are not adding big files which can impact the performance. And as someone making a request to add new collection, he/she should be aware of the expected size of file, instead of we add it and then later figure out by monitoring the logs.

File: pyServer/manifests/linux/linux-repoconfig [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ll,/etc/yum.repos.d/
+ll,/etc/apt
+
+echo,### Gathering Repository Configuration Files ###
+copy,/etc/products.d/*.prod,noscan
+copy,/etc/zypp/repos.d/*.repo,noscan
+copy,/etc/yum.repos.d/*.repo,noscan
+copy,/etc/apt/sources.list,noscan
+copy,/etc/apt/sources.list.d/*.list,noscan
+echo,
```
Comment (chintanrp) [inline pyServer/manifests/linux/linux-repoconfig:13]: Why do we need this "echo"?

**Comment by peden-msft** [comment]
@Chintan,

The specified files in /boot are just text configurations and should be very small.

I don't think /var/log/zypp/history should be any larger than the corresponding yum/dpkg/dnf files for other distributions.

I added the noscan parameter as requested on any files that should not contain credentials.

File: pyServer/manifests/linux/linux-repoconfig (hunk 2 above)
Comment (peden-msft) [reply pyServer/manifests/linux/linux-repoconfig:13]: I'm not sure, but it's on every other configuration block so I duplicated it here - in fact, I missed it in the bootconfig.  I'll fix that.

**Review by chintanrp — DISMISSED** [review]
LGTM

**Review by chintanrp — COMMENTED** [review]
The is an error for the build. Please fix as needed.

**Comment by peden-msft** [comment]
Is the issue just the missing newline?

File: docs/manifest_content.md [hunk 3]
```diff
@@ -1680,4 +1711,5 @@ workloadbackup | copy | /WindowsAzure/Logs/Plugins/\*
 workloadbackup | copy | /WindowsAzure/Logs/AggregateStatus/aggregatestatus\*.json
 workloadbackup | copy | /WindowsAzure/Logs/AppAgentRuntime.log
 
-*File was created by running [parse_manifest.py](../tools/parse_manifest.py) on `2021-08-24 11:59:47.343734`*
\ No newline at end of file
+*File was created by running [parse_manifest.py](../tools/parse_manifest.py) on `2021-08-24 11:59:47.343734`*
+
```
Comment (chintanrp) [inline docs/manifest_content.md:1715]: this is the extra line causing issue.

Comment (peden-msft) [reply docs/manifest_content.md:1715]: Can you please ping me on Teams?

**Review by chintanrp — APPROVED** [review]
LGTM

**Pull request merged**